eng/SourceBuild.Tarball.targets

```diff
@@ -0,0 +1,375 @@
+        Remove="@(MasterRepositoryDirToRemove)"
+        Condition="
+          $([System.Text.RegularExpressions.Regex]::IsMatch(
+            '%(Identity)',
+            '\.[A-Fa-f0-9]{40}$'))" />
+    </ItemGroup>
+
+    <RemoveDir Directories="@(MasterRepositoryDirToRemove)" />
+  </Target>
+
+  <Target Name="AddTarballSourcelinkMetadata">
+    <ItemGroup>
+      <ArtifactsGitDir Include="$(ClonedSubmoduleDirectory)**\.git\**\*" />
+    </ItemGroup>
+
+    <Copy
+      SourceFiles="@(ArtifactsGitDir)"
+      DestinationFiles="@(ArtifactsGitDir -> '$(TarballSourceDir)%(RecursiveDir)%(Filename)%(Extension)')" />
+  </Target>
+
+  <Target Name="AddTarballTools">
+    <ItemGroup>
+      <TarballRootToolFile
+        Include="
+          $(ProjectDir)*.proj;
+          $(ProjectDir)*.props;
+          $(ProjectDir)*.targets;
+          $(ProjectDir)global.json;
+          $(ProjectDir)DarcVersion.txt;
+          $(ProjectDir)ProdConFeed.txt;
+          $(ProjectDir)smoke-test*;
+          $(ProjectDir)eng\**;
+          $(ProjectDir)keys\**;
+          $(ProjectDir)patches\**;
+          $(ProjectDir)scripts\**;
+          $(ProjectDir)repos\**;
+          $(ProjectDir)tools-local\**;
+          "
+        Exclude="
+          $(ProjectDir)tools-local\arcade-services\**;
+          $(ProjectDir)tools-local\tasks\*\bin\**;
+          $(ProjectDir)tools-local\tasks\*\obj\**;
+          " />
+
+      <TarballRootToolFile
+        Include="$(DotNetCliToolDir)**"
+        Exclude="
+          $(DotNetCliToolDir)shared\2.1.0\**;
+          $(DotNetCliToolDir)tools\**;
+          "
+        Condition="'$(CustomDotNetSdkDir)' == ''"/>
+
+      <TarballCopyFile
+        Include="@(TarballRootToolFile)"
+        RelativeDestination="$([MSBuild]::MakeRelative('$(ProjectDir)', '%(Identity)'))" />
+
+      <TarballGitInfoFile Include="$(ProjectDir)artifacts\git-info\**" />
+      <TarballCopyFile
+        Include="@(TarballGitInfoFile)"
+        RelativeDestination="git-info\%(RecursiveDir)%(Filename)%(Extension)" />
+
+      <TarballBootstrappingToolFile Include="$(ProjectDir)support\tarball\tool-bootstrapping/**;" />
+      <TarballCopyFile
+        Include="@(TarballBootstrappingToolFile)"
+        RelativeDestination="tool-bootstrapping\%(RecursiveDir)%(Filename)%(Extension)" />
+
+      <TarballCopyFile
+        Include="$(ProjectDir)support/tarball/build.sh"
+        RelativeDestination="build.sh" />
+
+      <!-- Setup package version props to include both source-built and running PackageVersions.props -->
+      <TarballCopyFile
+        Include="$(ProjectDir)support\tarball\PackageVersions.props"
+        RelativeRoot="artifacts/obj/$(Platform)/Release/" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddTarballPackages">
+    <!-- Remove reference-packages from tarball prebuilts...-->
+    <ItemGroup Condition="'$(CustomRefPackagesDir)' != ''">
+      <ExcludePrebuiltPackageFile Include="$(CustomRefPackagesDir)**\*.nupkg" />
```

Comment:
I don't understand where `ExcludePrebuiltPackageFile` is used.  I only see it here and on line 177, but I don't see where it's used anyplace else.

Reply:
Good catch, this name is leftover from a refactor. It's meant to be the same item list as the task output. Fixed.